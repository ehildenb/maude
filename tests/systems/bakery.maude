

load ../../contrib/systems/bakery.maude

--- Variants of conditions in BAKERY-FVP

get variants in BAKERY-FVP : X:Nat < Y:Nat .

--- Reachability command

mod BAKERY-META-LEVEL is
   protecting BAKERY .
   protecting META-LEVEL .
endm

search [,2] s 0 ; s 0 ; [idle] [idle] =>* C:Conf .
