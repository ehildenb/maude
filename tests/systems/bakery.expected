==========================================
get variants in BAKERY-FVP : X < Y .

Variant #1
rewrites: 0
Bool: #1:Nat < #2:Nat
X --> #1:Nat
Y --> #2:Nat

Variant #2
rewrites: 2
Bool: true
X --> %1:Nat
Y --> %1:Nat + %2:NzNat

Variant #3
rewrites: 2
Bool: false
X --> %1:Nat + %2:Nat
Y --> %1:Nat

No more variants.
rewrites: 2
==========================================
search [, 2] in BAKERY-META-LEVEL : s ; s ; [idle] [idle] =>* C:Conf .

Solution 1 (state 0)
states: 1  rewrites: 0
C:Conf --> s ; s ; [idle] [idle]

Solution 2 (state 1)
states: 2  rewrites: 1
C:Conf --> (s s) ; s ; [idle] [wait(s)]

Solution 3 (state 2)
states: 3  rewrites: 2
C:Conf --> (s s s) ; s ; [wait(s)] [wait(s s)]

Solution 4 (state 3)
states: 4  rewrites: 3
C:Conf --> (s s) ; s ; [idle] [crit(s)]

No more solutions.
states: 4  rewrites: 3
Bye.
